app/src/main/java/de/westnordost/streetcomplete/quests/surface/SurfaceAnswer.kt

```diff
@@ -8,12 +8,16 @@ sealed interface SurfaceOrIsStepsAnswer
 object IsActuallyStepsAnswer : SurfaceOrIsStepsAnswer
 object IsIndoorsAnswer : SurfaceOrIsStepsAnswer
 
-data class SurfaceAnswer(val value: Surface, val note: String? = null) : SurfaceOrIsStepsAnswer
+data class SurfaceAnswer(val value: Surface, val note: String? = null, val replacesTracktype: Boolean = false) : SurfaceOrIsStepsAnswer
 
 fun SurfaceAnswer.applyTo(tags: Tags, key: String) {
     val osmValue = value.osmValue
     val previousOsmValue = tags[key]
 
+    if (replacesTracktype) {
+        tags.remove("tracktype")
+    }
+
```

Comment:
The previous values of the tags can be accessed here. Checking whether `tracktype` is in conflict with the selected surface answer and deleting it in that case rather belongs here than in the form, just like it is done for `smoothness`. There is no need to attach some boolean variable to the answer.

Reply:
I did this way to keep "should ask about tracktype conflict" and "should delete tracktype due to conflict" checks together and protect them from diverging.

Reply:
I understand. However, it is not the application logic's duty to know anything about what happens in the UI.

Reply:
In this case I see it as GUI collecting info from user - including confirmation that existing tracktype data should be thrown away